bob-backend/src/pearl/group.rs

```diff
@@ -10,14 +10,14 @@ use ring::digest::{digest, SHA256};
 
 #[derive(Clone, Debug)]
 pub struct Group {
-    holders: Arc<RwLock<Vec<Holder>>>,
+    holders: Arc<RwLock<HierarchicalBloom<Holder>>>,
```

Comment:
`HierarchicalBloom` works as container, thus it is better to reflect this in the name. Also, I think in the future we can include RangeFilter in this hierarchical structure. 

My suggestion for name: `HierarchicalFilteringContainer`